src/v/cloud_storage_clients/client_pool.cc

```diff
@@ -18,6 +18,7 @@
 #include <seastar/core/smp.hh>
```

Comment:
> Simplify the lease/borrow logic by never adding borrowed clients to the
local pool.

it would be nice if this commit message explained why things are simpler. the changes in the commit are complicated enough that it's not obvious that it is now some how simpler.

Reply:
As I have mentioned elsewhere, noted to myself to write less abstract commit messages in the future. Thank you for feedback! :thanks:

Reply:
thanks!